src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -260,6 +276,22 @@ public void periodic() {
                         chassisSpeeds.vyMetersPerSecond = yPID.calculate(getPose().getTranslation().minus(translationTarget).getY());
                 }
 
+                if(balanceTarget != null ) {
+                        if (Math.abs(getGyroscopePitch().getDegrees()) >= AUTO_BALANCE_TOLERANCE_DEGREES) {
+                                chassisSpeeds.vyMetersPerSecond = yPID.calculate(getGyroscopePitch().minus(balanceTarget).getRadians());
+                                lastUpdateTime = Timer.getFPGATimestamp();
+                        } else if (Math.abs(getGyroscopeRoll().getDegrees()) >= AUTO_BALANCE_TOLERANCE_DEGREES) {
+                                chassisSpeeds.vxMetersPerSecond = xPID.calculate(getGyroscopeRoll().minus(balanceTarget).getRadians());
+                                lastUpdateTime = Timer.getFPGATimestamp();
+                        }
+
+                        if(balanceTargetOriginalPose.getTranslation().getDistance(getPose().getTranslation()) > MAX_AUTO_BALANCE_TRANSLATION_METERS ) {
+                                balanceTarget = null;
```

Comment:
better to call setBalanceTarget method so that the drive speeds also get set to 0.